src/lib/exporter/videoExporter.ts

```
@@ -484,40 +549,123 @@ export class VideoExporter {
+						}
+					})
+					.catch((error) => {
+						if (!this.cancelled && !this.nativeEncoderError) {
+							this.nativeEncoderError =
+								error instanceof Error ? error : new Error(String(error));
+						}
+						if (!this.cancelled && !this.nativeWriteError) {
+							this.nativeWriteError =
+								error instanceof Error ? error : new Error(String(error));
+						}
+					}));
```

Biome (2.4.11)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)